### PR TITLE
Add OSSL_PROVIDER_do_all()

### PR DESCRIPTION
**Pull request by levitte**
This allows applications to iterate over all loaded providers.

### CODE REVIEW

**Comment by mattcaswell** [comment]
The travis errors are relevant

File: doc/man3/OSSL_PROVIDER.pod [hunk 1]
```diff
@@ -66,6 +69,10 @@ runs its teardown function.
 OSSL_PROVIDER_available() checks if a named provider is available
 for use.
 
+OSSL_PROVIDER_do_all() iterates over all loaded providers, calling
```
Comment (bernd-edlinger) [inline doc/man3/OSSL_PROVIDER.pod:72]: How about OSSL_PROVIDER_iterate_for_all or something?

the name ..._do_all() reminds me of the magic function
```C
 void do_what_i_mean(void);
```

Comment (bernd-edlinger) [reply doc/man3/OSSL_PROVIDER.pod:72]: but if you like _do_all I am okay with it as well :smile:

Comment (levitte) [reply doc/man3/OSSL_PROVIDER.pod:72]: `_do_all` is an already existing pattern

**Comment by levitte** [comment]
Documentation glitch fixed

**Comment by levitte** [comment]
Merged

a7ad40c502d3484e0a574e121cfba70631f224bf Add OSSL_PROVIDER_do_all()

**Pull request closed without merging**